python/pipeline/vreso.py

```diff
@@ -0,0 +1,175 @@
+import scanreader
+import numpy as np
+import datajoint as dj
+from pipeline.exceptions import PipelineException
+from pipeline import reso, experiment, mice, notify, shared
+from pipeline.utils import galvo_corrections, signal, quality, mask_classification, performance, h5
+
+# external storage place
+dj.config['external'] = {'protocol': 'file',
+                         'location': '/mnt/dj-stor01/pipeline-externals'}
+
+schema = dj.schema("pipeline_vreso", locals(), create_tables=False)
+CURRENT_VERSION = 1
+
+
+@schema
+class Amplifier(dj.Lookup):
+    definition = """ # Electrophysiology rig setup
+    amplifier                     : varchar(32)         # name and model of patch rig
+    """
+    contents = [
+        ['Axoclamp 2B'],
+        ['NPI ELC-03XS']
+    ]
+
+
+@schema
+class PatchSession(dj.Manual):
+    definition = """ # Single session of patch clamp experiment
+    -> mice.Mice
+    patch_session                 : smallint unsigned   # session index for the mouse
+    ---
+    -> Amplifier
+    -> experiment.Rig
+    -> experiment.Person
+    recording_path = ''           : varchar(256)        # path to folder containing all h5 files for the given session
+    session_ts=CURRENT_TIMESTAMP  : timestamp           # time of session
+    """
+
+
+@schema
+class Cell(dj.Manual):
+    definition = """ # Single cell patched during ephys experiment
+    -> PatchSession
+    cell_id                       : smallint unsigned   # cell index for patch session
+    ---
+    cell_notes = ''               : varchar(256)
+    """
+
+
+@schema
+class Recording(dj.Manual):
+    definition = """ # Single recording of a cell
+    -> Cell
+    recording_id                  : smallint unsigned   # recording index for patch session
+    ---
+    clamp_type                    : enum('voltage','current')
+    patch_type                    : enum('whole', 'loose')
```

Comment:
enums are generally discouraged because if you wanna add more values later, you would have to go to MySQL (and I thought it was not even possible but i just googled it and i think it is); if you don;t think there will be ny new types of clamp or patch, should be alright, otherwise you could add PatchType as a lookup (added benefit is that this lookup can have other properties like a description of what it means)

Reply:
There are only two types of clamps possible and patching is the same.